### PR TITLE
Fix double delete in RandGen

### LINKED ISSUE #255: Double delete in RandGen copy and move
**Describe the bug**
RandGen does not follow the rule of 5.
Default copy and more are used while we are manually destroying managed resources.

This will result in a double delete if copy or move is ever used with RandGen.

**To Reproduce**
Invoke copy or move on RandGen

**Expected behavior**
No double delete

**Additional context**
We should be using value semantics with RandGen now anyway.

### PR DESCRIPTION
**Pull request by lavinrp**
closes #255 
Requires #239 be closed first

Updates RandGen to use value semantics. This fixes some potential memory corruptions caused by using the default copy and move special members while owning raw pointers.

### CODE REVIEW

File: GameBackbone/Include/GameBackbone/Util/RandGen.h [hunk 1]
```diff
@@ -7,35 +7,50 @@
...
+		///		Throws if the provided min is not less than the max.
+		/// @param min The minimum number that can be returned.
+		/// @param max The maximum number that can be returned.
+		/// @throws std::runtime_error if min is greater than or equal to max
+		/// @return A random number between min and max
 		double uniDist(double min, double max);
 
 	protected:
 		// This is being made a pointer to remove the VS warning about declspecing seed.
 		// Everyone is going to have to compile this anyway with the same compiler as their project so they won't have issues.
```
Comment (gyurgyma) [inline GameBackbone/Include/GameBackbone/Util/RandGen.h:49]: This comment is now wrong.

**Review by gyurgyma — CHANGES_REQUESTED** [review]
I like this fix. Simple and easy. Definitely need to understand what the compiler was telling us. We should discuss that before merging this. Also we need to remove the comment that was wrong.

File: Tests/Source/RandGenTests.cpp [hunk 2]
```diff
@@ -10,28 +10,67 @@ BOOST_AUTO_TEST_SUITE(RandGen_CTRs)
...
+	std::string seedString = std::string("TestString");
+	RandGen testRandGen(seedString);
+
+	//Ensure that the seed is what we set;
+	BOOST_CHECK_EQUAL("TestString", testRandGen.getSeed());
+
+	// Ensure that the Distributor has been set up correctly
+	double output = testRandGen.uniDist(0, 1);
+	BOOST_CHECK(output >= 0 && output < 1);
+
```
Comment (gyurgyma) [inline Tests/Source/RandGenTests.cpp:64]: This isn't a copy. This comment is confusing. Let's go ahead and fix this naming and stuff.

File: GameBackbone/Include/GameBackbone/Util/RandGen.h (hunk 1 above)
Comment (lavinrp) [reply GameBackbone/Include/GameBackbone/Util/RandGen.h:49]: comment removed.

File: Tests/Source/RandGenTests.cpp (hunk 2 above)
Comment (lavinrp) [reply Tests/Source/RandGenTests.cpp:64]: renamed.

**Review by gyurgyma — APPROVED** [review]

**Pull request merged**